service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -243,4 +255,20 @@ private String buildPriorPrescription(MedicationRequest medicationRequest) {
         }
         return StringUtils.EMPTY;
     }
+
+    private String buildParticipant(MedicationRequest medicationRequest) {
+        Predicate<String> isPractitioner = s -> s.startsWith(ResourceType.Practitioner.name());
+        Predicate<String> isPractitionerRole = s -> s.startsWith(ResourceType.PractitionerRole.name());
+        Predicate<String> isOrganization = s -> s.startsWith(ResourceType.Organization.name());
```

Comment:
These aren't right, because `"PractitionerRole/UUID".startsWith("Practitioner")`

Reply:
to avoid it, there could be used, reference.getReferenceElement().getResourceType() and then verify if it is equal.